src/super_gradients/common/code_analysis/breaking_change.py

```diff
@@ -0,0 +1,268 @@
+                    line_num=source_class.lineno,
+                )
+            )
+
+    # FUNCTION SIGNATURES
+    source_functions_signatures = parse_functions_signatures(source_code)
+    current_functions_signatures = parse_functions_signatures(current_code)
+    for function_name, source_function_signature in source_functions_signatures.items():
+
+        if function_name in current_functions_signatures:
+            current_function_signature = current_functions_signatures[function_name]
+
+            # ParameterRemoved
+            for source_param in source_function_signature.params.all:
+                if source_param not in current_function_signature.params.all:
+                    breaking_changes.params_removed.append(
+                        ParameterRemoved(
+                            function_name=function_name,
+                            parameter_name=source_param,
+                            line_num=current_function_signature.line_num,
+                        )
+                    )
+
+            # RequiredParameterAdded
+            for current_param in current_function_signature.params.required:
+                if current_param not in source_function_signature.params.required:
+                    breaking_changes.required_params_added.append(
+                        RequiredParameterAdded(
+                            function_name=function_name,
+                            parameter_name=current_param,
+                            line_num=current_function_signature.line_num,
+                        )
+                    )
+
+        else:
+            # FunctionRemoved
+            breaking_changes.functions_removed.append(
+                FunctionRemoved(
+                    function_name=function_name,
+                    line_num=source_function_signature.line_num,
+                )
+            )
+
+    # Check import ONLY if __init__ file.
+    if module_path.endswith("__init__.py"):
+        source_imports = parse_imports(code=source_code)
+        current_imports = parse_imports(code=current_code)
+        breaking_changes.imports_removed = [
+            ImportRemoved(import_name=source_import, line_num=0) for source_import in source_imports if source_import not in current_imports
+        ]
+    return breaking_changes
+
+
+def analyze_breaking_changes(verbose: bool = 1) -> List[Dict[str, Union[str, List]]]:
+    """Analyze changes between the current branch (HEAD) and the master branch.
+    :param verbose: If True, print the summary of breaking changes in a nicely formatted way
+    :return:        List of changes, where each change is a dictionary listing each type of change for each module.
+    """
+    # GitHelper requires `git` library which should NOT be required for the other functions
+    from super_gradients.common.code_analysis.git_utils import GitHelper
+
+    root_dir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__)))))
```

Comment:
Not very clean... 
could you use 
```
from setuptools import find_packages
find_packages(where="./src")

```
maybe

Reply:
The issue is that this function can be called from anywhere, not just from the project root , so it's dangerous to work with the working directory path, which depends on where the script is called from. Using `find_packages` would not resolve it.
To make it robust to that, I chose ot work with path relatively to the file itself (__file__). 

But I agree it looks dirty.

Do you have an idea how to make it cleaner ?
We can maybe go for `root_dir = Path(__file__).resolve().parents[5]` which may be slightly better ? Not sure

Reply:
yes. even that is better

Reply:
0612555efc4a5b117f38d7ac8eb061bfa07e6b0e